test/tests/build/build_cmake_targets.vader

```diff
@@ -331,30 +354,197 @@ Execute (Check all targets for empty project):
 Execute (Select all targets for empty project):
     " enable tests for windows with last cmake
     if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
-        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt shouldn't be generated"
+        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should not be generated"
         let cmd = cmake4vim#SelectTarget('all')
         AssertEqual cmd, ''
-        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt shouldn't be generated"
+        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should not be generated"
     else
         echom 'SKIPPED'
     endif
 
 Execute (Call CMakeBuild for empty project):
     " enable tests for windows with last cmake
     if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
-        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt shouldn't be generated"
+        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should not be generated"
         silent CMakeBuild
-        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt shouldn't be generated"
+        Assert !filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should not be generated"
     else
         echom 'SKIPPED'
     endif
 
 Execute (Check CMakeClean for empty project):
     " enable tests for windows with last cmake
     if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
-        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+        Assert !isdirectory("cmake-build-Release"), "Build directory should not exist"
         CMakeClean
-        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+        Assert !isdirectory("cmake-build-Release"), "Build directory should not exist"
+    else
+        echom 'SKIPPED'
+    endif
+
+Execute (Check CMakeKit autocomplete):
+    if !has('win32')
+        let kits = split(cmake4vim#CompleteKit(0,0,0))
+        AssertEqual len(kits), 2
+        AssertEqual kits, [ "First", "Second" ]
+        let g:cmake_kits[ "Third" ] = { "toolchain_file" : "random_third.txt" }
+        let kits = split(cmake4vim#CompleteKit(0,0,0))
+        AssertEqual len(kits), 3
+        AssertEqual kits, [ "First", "Second", "Third" ]
+    else
+        echom 'SKIPPED'
+    endif
+
+Execute (Check CMakeKit toolchain file):
+    if !has('win32')
+        Assert !isdirectory("cmake-build-Release"), "Build directory should not exist"
+        silent CMakeSelectKit First
+        silent CMake
+        Assert isdirectory("cmake-build-Release"), "Build directory should exist"
+        " TODO: how to check cmake info to see if the toolchain file was supplied
```

Comment:
@ilyachur this and `CtrlP` integration is the only thing left.

Do you have any ideas for this?